ConsoleApp1/ConsoleApp1/Program.cs

```diff
@@ -0,0 +1,528 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.IO;
+using System.Windows.Input;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Snake
+{
+
+    public static class keyThread
+    {
+        public static List<ConsoleKey> keyBuffer = new List<ConsoleKey>();
+
+        public static void Trd()
+        {
+
+            while (true)
+            {
+                if (Console.KeyAvailable)
+                {
+                    ConsoleKey key = Console.ReadKey(true).Key;
+                    if (keyThread.keyBuffer.Count > 0)
+                    {
+                        if (keyThread.keyBuffer[keyThread.keyBuffer.Count - 1] != key)
+                            keyThread.keyBuffer.Add(key);
+                    }
+                    else
+                        keyThread.keyBuffer.Add(key);
+                }
+
+                Thread.Sleep(1);
```

Comment:
sleep 1 ms?

Reply:
Yep. by default this operation take 0.0003 ms. It's too many not functional operations per second without such step.